src/ResourceManager/Compute/Commands.Compute/VirtualMachine/Action/RestartAzureVMCommand.cs

```diff
@@ -67,6 +67,7 @@ public class RestartAzureVMCommand : VirtualMachineBaseCmdlet
            Position = 1,
            ValueFromPipelineByPropertyName = true,
            HelpMessage = "The virtual machine name.")]
+        [ResourceNameCompleter("Microsoft.Compute/virtualMachines", "ResourceGroupName")]
```

Comment:
@maddieclayton
 Can you move the `Name` Param into the `RestartResourceGroupNameParameterSet` and `PerformMaintenanceResourceGroupNameParameterSet` param set only? 

As of now if you specify the `resourceId` of the vm, the cmdlet still asks for the `Name` of the vm. Which is not what the users expect.

This might involve more changes [here](https://github.com/Azure/azure-powershell/blob/b2c9eead74d69b12aae236ca9102a36697928a0d/src/ResourceManager/Compute/Commands.Compute/VirtualMachine/Action/RestartAzureVMCommand.cs#L98), you'll have to get the name form the resourceId as well...

I observed the same behavior in `Remove-AzureRmVm` as well.

Reply:
Resolved offline, will be addressed in a different PR.